### PR TITLE
Publish JARs with deduped names.

### PR DESCRIPTION
**Pull request by a-honcharenko**
## What
Ability to consume each individual OSS module as it's own artifact.
Get rid of fat jars.

## How
- In root build.gradle add a new task subprojects can use. Any project that needs to publish can include this task in their build.gradle. The published artifact will use the subproject directory name as the artifact name.
- Rename any subprojects with identical names. Identically named subprojects have the following issues:
  -  publishing as is leads to classpath confusion when the jars with the same names are placed in the Java distribution. This leads to `NoClassDefFound` errors on runtime. 
  - deconflicting the jar names without changing directory names leads to dependency errors as the OSS jar pom files are generated using `project` dependencies (suggesting a dependency a sibling subproject in the same repo) that use subprojects `group` and `name` as a reference. This means the generated jars look for Jars that do not exists (as their names have been changed) and cannot compile.
  - the workaround to changing a subproject's name involves resetting the subproject's name in the `settings.gradle` and depending on the new name in each `build.gradle`. This increases configuration burden and decreases the ease of reading, since one will have to check the `settings.gradle` to know what the right subproject name is. See [this issue](https://github.com/gradle/gradle/issues/847) for more info.
  - Given that Gradle itself doesn't have support for identically named subprojects (see the linked issue), the simplest solution is to not allow duplicated directories. I've only renamed conflicting directories here to keep things simple. I will create a follow up issues to enforce non-identical subproject names in our builds.

Directories that are renamed:
* airybyte-config:models -> airbyte-config:config-models
* airybyte-config:persistence -> airbyte-config:config-persistence.
* airbyte-db:lib to airbyte-db:db-lib.
* airbyte-metrics:lib to airbyte-metrics:metrics-lib.
* airbyte-protocol:models -> airbyte-protocol:protocol-models
* airbyte-scheduler:models -> airbyte-scheduler:scheduler-models.
* airbyte-scheduler:persistence -> airbyte-scheduler:scheduler-persistence

This will be consumed by https://github.com/airbytehq/airbyte-cloud/pull/1771. Take a look at the various build.gradles in this PR to understand how the dependencies are used.

## Recommended reading order
1. top-level `build.gradle` - to understand the new task.
2. `<subproject>/build.gradle` - to understand how to turn on publishing and which projects are publishing.
3. Everything else is renaming and including the new subproject names.

### CODE REVIEW

File: build.gradle [hunk 1]
```diff
@@ -18,9 +18,12 @@ plugins {
     id 'com.github.hierynomus.license' version '0.16.1'
     id 'com.github.spotbugs' version '5.0.5'
     id 'version-catalog'
+    id 'java-library'
     id 'maven-publish'
 }
 
+apply from: "$rootDir/repositories.gradle"
```
Comment (davinchia) [inline build.gradle:25]: this is a nice touch!

File: build.gradle [hunk 2]
```diff
@@ -18,9 +18,12 @@ plugins {
     id 'com.github.hierynomus.license' version '0.16.1'
     id 'com.github.spotbugs' version '5.0.5'
     id 'version-catalog'
+    id 'java-library'
```
Comment (davinchia) [inline build.gradle:21]: why do we need this java-library plugin here?

**Review by davinchia — APPROVED** [review]
Very nice! One question on a plugin, otherwise we are good to go!

File: build.gradle (hunk 2 above)
Comment (a-honcharenko) [reply build.gradle]: After testing options I forgot to remove it.
Pushed changes.

**Review by jdpgrailsdev — APPROVED** [review]
:shipit:  I only reviewed the Gradle file changes and all looks well.

**Comment by davinchia** [comment]
This PR was split into the following PR for better change management:
- #13427 
- #13476 
- #13478 
- #13480 
- #13484

**Pull request closed without merging**